app/src/main/java/app/musikus/services/SessionService.kt

```diff
@@ -70,27 +68,31 @@ data class NotificationActionButtonConfig(
     val tapIntent: PendingIntent?
 )
 
+
+
 @AndroidEntryPoint
 class SessionService : Service() {
 
+    @Inject
+    @ApplicationScope
+    lateinit var applicationScope: CoroutineScope
```

Comment:
if we make the `onEvent` function a suspend function, we don't need the application scope inside the service.

Reply:
Unfortunately, we need applicationScope also in `onStartCommand` when calling `suspend createNotification()`.

Reply:
Fair enough. It probably is a good idea to run the coroutines in the application scope rather than in the activesessionviewmodel scope, so they won't get canceled if the active session is closed.